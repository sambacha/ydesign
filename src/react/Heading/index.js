import React from 'react'
import PropTypes from 'prop-types'
import Box from '../Box'
import theme from '../theme'

const defaultProps = {
  fontWeight: 3,
  lineHeight: 'title',
}

const Heading = React.forwardRef((props, ref) => <Box ref={ref} {...props} />)

Heading.defaultProps = {
  ...defaultProps,
  as: 'h2',
  fontSize: '',
  theme,
}

Heading.propTypes = {
  ...Box.propTypes,
  theme: PropTypes.object,
}

Heading.displayName = 'Heading'

// H1
Heading.h1 = props => <Heading {...props} />
Heading.h1.defaultProps = {
  ...defaultProps,
  as: 'h1',
  fontSize: 6,
}
Heading.h1.displayName = 'Heading.h1'

// H2
Heading.h2 = props => <Heading {...props} />
Heading.h2.defaultProps = {
  ...defaultProps,
  as: 'h2',
  fontSize: 5,
}
Heading.h2.displayName = 'Heading.h2'

// H3
Heading.h3 = props => <Heading {...props} />
Heading.h3.defaultProps = {
  ...defaultProps,
  as: 'h3',
  fontSize: 4,
}
Heading.h3.displayName = 'Heading.h3'

// H4
Heading.h4 = props => <Heading {...props} />
Heading.h4.defaultProps = {
  ...defaultProps,
  as: 'h4',
  fontSize: 3,
}
Heading.h4.displayName = 'Heading.h4'

// H5
Heading.h5 = props => <Heading {...props} />
Heading.h5.defaultProps = {
  ...defaultProps,
  as: 'h5',
  fontSize: 2,
}
Heading.h5.displayName = 'Heading.h5'

// H6
Heading.h6 = props => <Heading {...props} />
Heading.h6.defaultProps = {
  ...defaultProps,
  as: 'h6',
  fontSize: 1,
}
Heading.h6.displayName = 'Heading.h6'

export default Heading
