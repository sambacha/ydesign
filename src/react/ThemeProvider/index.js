import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
