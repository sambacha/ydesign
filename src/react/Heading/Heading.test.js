import Heading from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Heading component sanity', () => {
  it('has name', () => {
    expect(Heading.displayName).toBe('Heading');
  });

  it('matches default snapshot', () => {
    const component = render(<Heading as={'h1'}>Heading 1</Heading>);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Heading as={'h1'}>Heading 1</Heading>
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
