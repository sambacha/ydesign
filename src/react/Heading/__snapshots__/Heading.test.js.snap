// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component sanity matches default snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  font-weight: 600;
  line-height: 1.25;
}

<h1
  class="c0"
  font-size=""
  font-weight="3"
>
  Heading 1
</h1>
`;

exports[`Heading component sanity matches themed snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  font-weight: 600;
  line-height: 1.25;
}

<h1
  class="c0"
  font-size=""
  font-weight="3"
>
  Heading 1
</h1>
`;
